### DISCUSSION TITLE
Zephyr port

### DISCUSSION THREAD

**Discussion by jgromes** [jgromes/RadioLib#1194] [Q&A]
Hii,

Is it possible to port this application to Zephyr? What challenges and technical considerations would be involved in the process?

I need to do a LoRA 2.4GHz application, using SX1280 transceiver.

_Originally posted by @sluissantos in https://github.com/jgromes/RadioLib/discussions/209#discussioncomment-10407768_

**Comment by jgromes** [comment]
@sluissantos I recommend you check out the port guide on the Wiki: https://github.com/jgromes/RadioLib/wiki/Porting-to-non-Arduino-Platforms

As to the specific challenges - I don't know, I'm not familiar with Zephyr. You will need to write the hardware abstraction layer that takes care of things like SPI transacations. But anything above that should work.

**Reply by HeadBoffin** [reply]
Zephyr, being Arduino on steroids, has all the hardware abstraction you can eat. But I'd recommend that the OP uses RL's SX1280 driver as the basis of writing a native Zephyr driver for SX1280 as that's how they roll there. There are already drivers for other SX radios  in Zephyr as a starting point.

**Comment by sluissantos** [comment]
Much thanks for your answers!!
